src/compiler/binder.rs

```diff
@@ -510,48 +509,39 @@ impl BinderType {
     }
 
     fn bind_worker(&self, node: &Node) {
-        match &*node {
-            Node::TypeParameterDeclaration(type_parameter_declaration) => {
-                self.bind_type_parameter(type_parameter_declaration)
-            }
-            Node::ParameterDeclaration(parameter_declaration) => {
-                self.bind_parameter(parameter_declaration)
-            }
-            Node::VariableDeclaration(variable_declaration) => {
-                self.bind_variable_declaration_or_binding_element(variable_declaration)
-            }
-            Node::TypeElement(TypeElement::PropertySignature(property_signature)) => {
-                self.bind_property_worker(property_signature)
-            }
-            Node::PropertyAssignment(property_assignment) => self
-                .bind_property_or_method_or_accessor(
-                    property_assignment,
-                    SymbolFlags::Property,
-                    SymbolFlags::PropertyExcludes,
-                ),
-            Node::Statement(Statement::FunctionDeclaration(function_declaration)) => {
-                self.bind_function_declaration(function_declaration)
+        match node {
+            Node::TypeParameterDeclaration(_) => self.bind_type_parameter(node),
```

Comment:
Now that I'm always passing `&Node`, a lot of these can probably be `match`'s on `node.kind()` (closer to the Typescript version)